internal/config/server.go

```diff
@@ -67,9 +71,19 @@ func NewServerConfig() (*ServerConfig, error) {
 		logger = NewLogger(os.Stdout, envs.LogLevel)
 	}
 
-	return &ServerConfig{
+	cfg := &ServerConfig{
 		Log:             logger,
 		Envs:            envs,
 		ShutdownTimeout: defaultShutdownTimeout,
-	}, nil
+	}
+
+	if cfg.Envs.D3SN != "" {
+		db, err := cfg.InitPostgresDB(cfg.Envs.D3SN)
```

Comment:
In mem.go I am not initializing the connection I am just passing the DB object (pool) to the store so it can use the connection pool. I thought to initialize the DB as early as possible as it might be one of the weak points in the application and just store the Pool as a config option so it can be shared with other parts of the app if needed.

Reply:
I am not sure, if it is a good idea: the initialization logic belongs to the DB package, so you should probably move it there. You can initialize the object in `main()` or somewhere close to the entry point and then inject it into any components, that use the DB connection. It seems that placing the DB handler into a configuration object so any component can use it is a bad practice: you want to control which components can send requests to your storage

Reply:
Yeah, it makes sense. I moved the DB logic into a database package and then initialised the db within the main function so that it allows me to share the db pool with any resource that should need it.